openpype/plugins/publish/collect_frame_range_data.py

```diff
@@ -0,0 +1,85 @@
+import pyblish.api
+import clique
+from openpype.pipeline import publish
+from openpype.lib import BoolDef
+
+
+class CollectFrameRangeData(pyblish.api.InstancePlugin,
+                            publish.OpenPypePyblishPluginMixin):
+    """Collect Frame Range data.
+    """
+
+    order = pyblish.api.CollectorOrder + 0.491
+    label = "Collect Frame Range Data"
+    families = ["plate", "pointcache",
+                "vdbcache", "online",
+                "render"]
+    hosts = ["traypublisher"]
+
+    def process(self, instance):
+        repres = instance.data.get("representations")
+        asset_data = None
+        if repres:
+            first_repre = repres[0]
+            if "ext" not in first_repre:
+                self.log.warning("Cannot find file extension"
+                                 " in representation data")
+                return
+
+            files = first_repre["files"]
+            collections, remainder = clique.assemble(files)
+            if not collections:
+                # No sequences detected and we can't retrieve
+                # frame range
+                self.log.debug(
+                    "No sequences detected in the representation data."
+                    " Skipping collecting frame range data.")
+                return
+            collection = collections[0]
+            repres_frames = list(collection.indexes)
+            asset_data = {
+                "frameStart": repres_frames[0],
+                "frameEnd": repres_frames[-1],
+            }
+
+        else:
+            self.log.info(
+                "No representation data. Using Asset Entity data instead")
+            asset_doc = instance.data.get("assetEntity")
+
+            attr_values = self.get_attr_values_from_data(instance.data)
+            if attr_values.get("setAssetFrameRange", True):
+                if instance.data.get("frameStart") is not None or not asset_doc:            # noqa
+                    self.log.debug("Instance has no asset entity set."
+                                   " Skipping collecting frame range data.")
+                    return
```

Comment:
This logged message will be confusing for the cases where `frameStart` is None but an asset document is available. I'd say we should either just check the `asset_doc` and assume `frameStart` will be available in instance data or validate separately with a clearer logged message?